maven_iuvs/echelle.py

```diff
@@ -2298,55 +2365,54 @@ def lineshape_model(params, wavelength_data, binedges, theCLSF, BU_bg):
               the cumulative line spread function for the LSF
     BU_bg : array
             An alternate background, constructed as described in Mayyasi+2023. 
+    fit_IPH_component : bool
+         whether to fit an IPH component to this spectrum in addition to Mars H and D
 
     Returns:
     ----------
     I_bin : array
             brightness per bin 
     """
-    total_brightness_H = params[0] # DN
-    total_brightness_D = params[1] 
-    total_brightness_IPH = params[2] 
-    central_wavelength_H = params[3] # nm
-    central_wavelength_D = params[3] - D_offset # nm
-    central_wavelength_IPH = params[4]  # TODO: Can calculate based on 
-                                         # spacecraft ephemeris and pass in for 
-                                         # the starting guess/requirement.
-    width_IPH = params[5]  # TODO: Should also be able to calculate this, 
-                            # which is the width of the Gaussian. 
+    param_dict = make_fit_param_dict(params, BU_bg=BU_bg)
```

Comment:
nice, this is more elegant. Changing this function to a class with static class members for the names / bounds / etc. is probably the move to avoid having things at the global scope, but the current implementation is good enough.